### ISSUE TITLE
autoscale differences between 1.5.1 and 2.0.0b3

### ISSUE THREAD

**Issue by LevN0** [matplotlib/matplotlib#6968]
Based on description of tight ("If True, set view limits to data limits"), I would have expected MPL 1.5.1 behavior (see code and image below).  Though perhaps there was a little margin added intentionally? 

Edit: I see this first part was intended, as change https://github.com/matplotlib/matplotlib/pull/5583. See second issue below.

```
import matplotlib as mpl

plt.plot([1,2,3,4])
plt.autoscale(axis='both', tight=True)

plt.show()
```

![limits_tight](https://cloud.githubusercontent.com/assets/15056721/17906688/62f63912-6946-11e6-9a96-baaea8a0b64c.png)

---

Aside from this, MPL 1.x use to conveniently adjust view limits for log and symlog when tight autoscaling was not on, but MPL 2 does not seem to do this anymore. 

This actually also looks like a result of https://github.com/matplotlib/matplotlib/pull/5583. I would argue the previous behavior was actually desirable in many instances. Is there a way to recover it? 

```
import matplotlib as mpl

plt.plot([1,2,3,4])

ax = plt.gca()
ax.set_xscale('log')
ax.set_yscale('log')

plt.show()
```

![log_scaling](https://cloud.githubusercontent.com/assets/15056721/17906913/4436397c-6947-11e6-947a-783118789c9d.png)

Python 2.7 and 3.4
Using tkagg backend.
Windows 7

**Comment by efiring** [comment]
@LevN0 in answer to your question, with v2.x if you start with `mpl.style.use('classic')` you will get the same autoscaling as with 1.5.  You can also use a context manager, or you can just set the relevant rcParams:
```python
mpl.rcParams['axes.autolimit_mode'] = 'round_numbers'
mpl.rcParams['axes.xmargin'] = 0
mpl.rcParams['axes.ymargin'] = 0
```

**Comment by NelleV** [comment]
Closed in #7458

**Comment by LevN0** [comment]
@efiring Thank you. I tried axes.autolimit_mode and it seems to result in the MPL 1.5 behavior for that aspect recovered.
